react/features/device-selection/actions.js

```diff
@@ -1,18 +1,173 @@
-/* globals APP */
+/* globals APP, interfaceConfig */
 
 import { openDialog } from '../base/dialog';
 import JitsiMeetJS from '../base/lib-jitsi-meet';
+import { API_ID } from '../../../modules/API/constants';
+import {
+    setAudioInputDevice,
+    setAudioOutputDevice,
+    setVideoInputDevice
+} from '../base/devices';
+import { i18next } from '../base/i18n';
+import {
+    PostMessageTransportBackend,
+    Transport
+} from '../../../modules/transport';
 
+import { SET_DEVICE_SELECTION_POPUP } from './actionTypes';
 import { DeviceSelectionDialog } from './components';
 
+
 /**
- * Open DeviceSelectionDialog with a configuration based on the environment's
- * supported abilities.
+ * Sets information about device selection popup in the store.
+ *
+ * @param {Object} popupDialog - Information about the popup.
+ * @param {Object} popupDialog.popup - The popup object returned from
+ * window.open.
+ * @param {Object} popupDialog.transport - The transport instance used for
+ * communication with the popup window.
+ * @returns {{
+ *     type: SET_DEVICE_SELECTION_POPUP,
+ *     popupDialog: Object
+ * }}
+ */
+function _setDeviceSelectionPopup(popupDialog) {
+    return {
+        type: SET_DEVICE_SELECTION_POPUP,
+        popupDialog
+    };
```

Comment:
As we discussed off-github can we rename this to popupDialogData or something similar that would give us an indication that this isn't a component, but some backend instead.